### DISCUSSION TITLE
Using app_tables and server calls

### DISCUSSION THREAD

**Discussion by lewysigns** [anvilistas/tabulator#73] [Q&A] [answered]
Is there a way to set Tabulator that is using the ```app_tables``` option to search with a given server function instead of an open .search() by default?

I am trying to achieve the following:

Define a query based on users role/permission. This is already done in a server function and don't know how this would be done (securely) on the client side.

**EDIT**:
This would allow to set our data tables as only accessible on the server side.

**Comment by s-cork** [comment, marked as answer]
Have you tried sending a client readable view of the table based on a certain role?

That would be the easiest fix. That way the app table you send to the client would already have restrictions.

**Reply by lewysigns** [reply]
Yes, when I send over the object returned by an app_table query, the tabulator seems to have issues. Here is a link demonstrating the issue:

https://anvil.works/build#clone:TX5IJADW5BLXJTRB=IGU6N4GJMVJGZXZHMS4NRCBH

I have been using anvil_extras serialisation library to get around this issue, but that is slowing down my request quite a bit as my data grows (4 seconds).

**Reply by s-cork** [reply]
When using the app_table option you don't set the data. You set the `self.tabulator.options['app_table'] = anvil.server.call(...)`
 

Have a look at the readme under the App Tables section.

**Reply by lewysigns** [reply]
Thank you for explaining. I definitely read the ReadMe, but it wasn't clear to me that you could set the "app_table" option to an anvil server call. That is exactly what I was looking for! 

You explicitly say "**You may want to make a server call to retrieve a client_readable view of an app_table**" yet my brain couldn't put it together to place the server call in the options definition!

Sorry for the post :)
